File: sdkwork-ai-sdk-java/src/main/java/com/sdkwork/ai/api/Responses2Api.java
package com.sdkwork.ai.api;

import com.sdkwork.ai.http.HttpClient;
import com.sdkwork.ai.model.*;
import java.util.List;
import java.util.Map;

public class Responses2Api {
    private final HttpClient client;
    
    public Responses2Api(HttpClient client) {
        this.client = client;
    }

    /** Cancel response */
    public CancelResponsePostResponse cancelResponse(String response_id) throws Exception {
        return (CancelResponsePostResponse) client.post(ApiPaths.aiPath("/v1/responses/" + response_id + "/cancel"), null);
    }

    /** Create response */
    public CreateResponseResponse createResponse(CreateResponseRequest body) throws Exception {
        return (CreateResponseResponse) client.post(ApiPaths.aiPath("/v1/responses"), body);
    }

    /** List response input items */
    public ListResponseInputItemsResponse listResponseInputItems(String response_id) throws Exception {
        return (ListResponseInputItemsResponse) client.get(ApiPaths.aiPath("/v1/responses/" + response_id + "/input_items"));
    }

    /** Get response */
    public GetResponseResponse getResponse(String response_id) throws Exception {
        return (GetResponseResponse) client.get(ApiPaths.aiPath("/v1/responses/" + response_id + ""));
    }

    /** Delete response */
    public DeleteResponseResponse deleteResponse(String response_id) throws Exception {
        return (DeleteResponseResponse) client.delete(ApiPaths.aiPath("/v1/responses/" + response_id + ""));
    }
}

File: sdkwork-ai-sdk-kotlin/src/main/kotlin/com/sdkwork/ai/api/Responses2Api.kt
package com.sdkwork.ai.api

import com.sdkwork.ai.*
import com.sdkwork.ai.http.HttpClient

class Responses2Api(private val client: HttpClient) {

    /** Cancel response */
    suspend fun cancelResponse(response_id: String): CancelResponsePostResponse? {
        return client.post(ApiPaths.aiPath("/v1/responses/$response_id/cancel"), null) as? CancelResponsePostResponse
    }

    /** Create response */
    suspend fun createResponse(body: CreateResponseRequest): CreateResponseResponse? {
        return client.post(ApiPaths.aiPath("/v1/responses"), body) as? CreateResponseResponse
    }

    /** List response input items */
    suspend fun listResponseInputItems(response_id: String): ListResponseInputItemsResponse? {
        return client.get(ApiPaths.aiPath("/v1/responses/$response_id/input_items")) as? ListResponseInputItemsResponse
    }

    /** Get response */
    suspend fun getResponse(response_id: String): GetResponseResponse? {
        return client.get(ApiPaths.aiPath("/v1/responses/$response_id")) as? GetResponseResponse
    }

    /** Delete response */
    suspend fun deleteResponse(response_id: String): DeleteResponseResponse? {
        return client.delete(ApiPaths.aiPath("/v1/responses/$response_id")) as? DeleteResponseResponse
    }
}
